Allow overriding visual look in server settings

lubomir opened the pull request:
Two new keys can be specified in the server settings JSON file.

 * logo - a path to image file that should be added to header
 * customStyle - a path to stylesheet that should be included on the page

There should be no change in how the UI looks, but it should allow us to customize it to match header of our PDC instance. See `redhat-style` branch in my fork for how that could be done.

lubomir committed:
Allow overriding visual look in server settings

Two new keys can be specified in the server settings JSON file.

 * logo - a path to image file that should be added to header
 * customStyle - a path to stylesheet that should be included on the
   page

hluk commented on src/app.jsx:
```diff
@@ -405,31 +407,43 @@ module.exports = React.createClass({
+      var logo = null;
+      var style = null;
+      if (localStorage['logo']) {
+          logo = <img src='/redhat-logo.png' className="brand-logo" />;
```
This shouldn't contain `redhat`.

hluk commented:
Will the stylesheet and logo be part of internal ansible repos?

zhutaoredhat commented:
http://www.patternfly.org/pattern-library/navigation/vertical-navigation/#/design
http://www.patternfly.org/pattern-library/navigation/horizontal-navigation/
@hluk  We could use the style of PatternFly navigation.

lubomir committed:
Add optional links to top right corner

The server settings can define a list of links that will be added to the
menu.

lubomir committed:
Use PatternFly for menu

The header is pretty much similar to the server UI, except for the
missing User menu in top right corner and the navigation bar being
empty.

lubomir commented:
Changes since previous version:
* by default the header looks the same as what PDC server itself does (using Patternfly)
* it's possible to customize the header with new options in server settings file (internal branding will live in ansible repo, PR is in progress)

hluk commented on src/app.jsx:
```diff
@@ -162,6 +166,9 @@ module.exports = React.createClass({
       localStorage.setItem('server', data['server']);
+      self.state.logo = data['logo'] || null;
+      self.state.customStyle = data['customStyle'] || null;
+      self.state.links = data.links;
```
Can you also add some defaults or just comments for the options in serversettings.json?

lubomir replied:
Good point. I can't add comments to the default JSON file. But documenting it in README might work as well.

hluk replied:
> I can't add comments to the default JSON file.

Oh, yeah, I forgot how much JSON sucks :).

lubomir committed:
Add documentation for customization to README

hluk commented:
Sorry, I just pushed some conflicting code to master.

lubomir committed:
Add url-loader dev dependency

Our webpack config uses this, but we originally did not install it (as
the path that needed it was never excercised).
